validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -36,67 +35,114 @@
                              ['batch', 'required', 'preserve'])
 
 
-class PredecessorTreeNode:
-    def __init__(self, children=None, readers=None, writer=None):
-        self.children = children if children is not None else {}
-        self.readers = readers if readers is not None else []
-        self.writer = writer
+class Node:
+    def __init__(self, data=None):
+        self.data = data
+        self.children = {}
 
     def __repr__(self):
-        retval = {}
+        return repr({
+            'data': self.data,
+            'children': self.children
+        })
 
-        if self.readers:
-            retval['readers'] = self.readers
-        if self.writer is not None:
-            retval['writer'] = self.writer
-        if self.children:
-            retval['children'] = \
-                {k: literal_eval(repr(v)) for k, v in self.children.items()}
 
-        return repr(retval)
-
-
-class PredecessorTree:
-    def __init__(self, token_size=2):
+class Tree:
```

Comment:
Calling this a Tree is confusing, since its interface doesn't follow expected Tree semantics. Call this a RadixTree or AddressedTree or something better.